parse.go

```diff
@@ -57,6 +104,18 @@ func (*Parser) Parse(parent ast.Node, block text.Reader, pctx parser.Context) as
 	return &n
 }
 
+func endOfObsidianHashtag(line []byte) int {
+	gr := uniseg.NewGraphemes(string(line))
```

Comment:
Minor:
This will convert every line that has a `#` into a string and attempt to walk it with uniseg.
I think we can optimize this ever so slightly by cutting the string at the first space, if any, since hashtags definitely do not contain spaces under either variant.